main.go

```diff
@@ -234,24 +222,29 @@ func (p *Propagator) JobUpdates(extID string, maxRetries int64) ([]DBJobStatusUp
 			&r.CreatedDate,
 		)
 		if err != nil {
-			p.rollback = true
 			return nil, err
 		}
 		retval = append(retval, r)
 	}
 	err = rows.Err()
-	if err != nil {
-		p.rollback = true
-	}
 	return retval, err
 }
 
 // MarkPropagated marks the job as propagated in the database as part of the
 // transaction tracked by the *Propagator.
 func (p *Propagator) MarkPropagated(id string) error {
+	tx, err := p.db.Begin()
+	if err != nil {
+		tx.Rollback()
```

Comment:
I'm iffy on this Rollback() call.

Reply:
I'm relatively certain that `tx` will be  nil if an error occurs in `Begin`. You can omit these rollbacks.